src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.Context.cs

```diff
@@ -29,26 +28,34 @@ public partial class SslStreamTests
         [BenchmarkCategory(Categories.NoAOT)]
         public Task DefaultHandshakeContextIPv6Async() => DefaultContextHandshake(_clientIPv6, _serverIPv6);
 
-        private async Task DefaultContextHandshake(Stream client, Stream server)
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public Task DefaultMutualHandshakeContextIPv4Async() => DefaultContextHandshake(_clientIPv4, _serverIPv4, true);
+
+        [Benchmark]
+        [BenchmarkCategory(Categories.NoAOT)]
+        public Task DefaultMutualHandshakeContextIPv6Async() => DefaultContextHandshake(_clientIPv6, _serverIPv6, true);
+
+        private async Task DefaultContextHandshake(Stream client, Stream server, bool requireClientCert = false)
         {
             if (_context == null)
             {
                 _context = SslStreamCertificateContext.Create(_cert, null);
             }
-            
+
             SslServerAuthenticationOptions serverOptions = new SslServerAuthenticationOptions
             {
                 AllowRenegotiation = false,
                 EnabledSslProtocols = SslProtocols.None,
                 CertificateRevocationCheckMode = X509RevocationMode.NoCheck,
-                ServerCertificateContext  = _context,
+                ServerCertificateContext = _context,
             };
 
             using (var sslClient = new SslStream(client, leaveInnerStreamOpen: true, delegate { return true; }))
             using (var sslServer = new SslStream(server, leaveInnerStreamOpen: true, delegate { return true; }))
             {
                 await Task.WhenAll(
-                    sslClient.AuthenticateAsClientAsync("localhost", null, SslProtocols.None, checkCertificateRevocation: false),
+                    sslClient.AuthenticateAsClientAsync("localhost", requireClientCert ? new X509CertificateCollection() { _clientCert } : null, SslProtocols.None, checkCertificateRevocation: false),
```

Comment:
just for my education: Why do you want to create a new instance of `X509CertificateCollection` every time the benchmark is called? Does this reflect typical real-life scenario?

Reply:
I wanted to mirror what we do in non-context benchmarks for SslStream.

https://github.com/dotnet/performance/blob/b69d1bdd939f8ccbebf62cd276d3f3d95149b1da/src/benchmarks/micro/libraries/System.Net.Security/SslStreamTests.cs#L130

Typical real-world scenarios would probably always use the overload accepting Ssl(Client|Server)AuthenticationOptions and reuse a single instance of it for multiple connections, or always create new instance entirely from scratch (as we do in linked source code).